server/services/messaging_service.go

```diff
@@ -303,7 +303,14 @@ func (s *messagingServerImpl) CreateBlurb(ctx context.Context, in *pb.CreateBlur
 	id := puid.Next()
 	name := fmt.Sprintf("%s/blurbs/%d", parent, id)
 	now := ptypes.TimestampNow()
-
+	if len(b.LegacyRoomId) >0 {
```

Comment:
Go has this cool way of declaring/initializing a variable and then checking it in the scope of a conditional, we might want to use that here.

```go
if roomId := b.GetLegacyRoomId(); roomId != "" {
  name = fmt.Sprintf("%s/blurbs/%d.%s", parent, id, roomId)
}
```

Same goes for the `userId` conditional

Reply:
Sounds cool! done.